transformers/DsChiefTransformer.js

```diff
@@ -22,39 +22,50 @@ module.exports = (address, nameSuffix = '') => ({
   },
 });
 
-const vdQuery = `SELECT *
+const chiefQuery = `SELECT *
     FROM dschief.lock
     WHERE tx_id = $1
     AND log_index = $2`;
 
+const delegateQuery = `SELECT *
+    FROM dschief.delegate_lock
+    WHERE tx_id = $1
+    AND log_index = $2`;
+
 const handlers = {
   'free(uint256)': async (services, { log, note }) => {
     const tx = await getTxByIdOrDie(services, log.tx_id);
 
     //get delegate event
     try {
-      const transaction = await services.provider.getTransaction(tx.hash);
-      //get transaction receipt
-      const { logs } = await transaction.wait();
-      //get event(s) from Delegate contract. Usually only one per transaction
-      const delegateLogArray = logs.filter(l => l.topics[0] === FreeTopic);
-      delegateLogArray.forEach(async delegateLog => {
-        await insertDelegateLock(services, {
-          fromAddress: tx.from_address,
-          immediateCaller: '0x' + delegateLog.topics[1].slice(-40),
-          lock: new BigNumber(delegateLog.data).div(new BigNumber("1e18")).negated().toString(),
-          contractAddress: delegateLog.address.toLowerCase(),
-          txId: log.tx_id,
-          blockId: log.block_id,
-          logIndex: delegateLog.logIndex,
-        });
-      });
+      const delegateRow = await services.db.oneOrNone(delegateQuery, [log.tx_id, log.log_index]);
```

Comment:
I'm wondering if we need to call this for each delegate log in the `delegateLogArray`. Reasoning being that a specific lock or free call could be called more than once from another delegate event if they are all in the same transaction.
Although it may not be necessary if that is caught by the catch block

Reply:
Good catch, you're right this is checking the wrong log index, will update